### PR TITLE
Add support for custom user data.

### PR DESCRIPTION
**Pull request by pmeijer**
Available REST end-points.
- get/put/patch/delete: 'user/data'
- get/put/patch/delete: 'users/:username/data'

(See generated REST doc for more info.)

### CODE REVIEW

File: src/server/api/index.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
 /**
  * @module Server:API
  * @author lattmann / https://github.com/lattmann
+ * @author pmeijer / https://github.com/pmeijer
+ * @author kesco / https://github.com/kesco
```
Comment (lattmann) [inline src/server/api/index.js]: typo: kesco should read kecso

File: src/server/api/webgme-api-user-data-patch.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "pluingSettings": {
+    "MyPlugin1": {
+      save: true
```
Comment (lattmann) [inline src/server/api/webgme-api-user-data-patch.json]: missing double quotes.

File: src/server/api/webgme-api-user-data-retrieve.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "pluingSettings": {
+    "MyPlugin1": {
+      enable: true,
```
Comment (lattmann) [inline src/server/api/webgme-api-user-data-retrieve.json]: missing double quotes.

File: src/server/api/webgme-api-user-data-retrieve.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+      save: false
```
Comment (lattmann) [inline src/server/api/webgme-api-user-data-retrieve.json]: missing double quotes.

File: src/server/api/webgme-api-user-data-retrieve.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    },
+    "MyPlugin2": {
+      outputName: "afile.ext"
```
Comment (lattmann) [inline src/server/api/webgme-api-user-data-retrieve.json]: missing double quotes.

File: src/server/api/index.js [hunk 6]
```diff
@@ -268,19 +352,66 @@ function createAPI(app, mountPath, middlewareOpts) {
...
     router.patch('/users/:username', function (req, res, next) {
+        // body params
+        //"email": "user@example.com",
+        //"password": "pass",
+        //"canCreate": null,
+        //"siteAdmin": false,
+        //"data": {}
+        ensureSameUserOrSiteAdmin(req, res)
+            .then(function (userData) {
+                if (req.body.hasOwnProperty('siteAdmin') && !userData.siteAdmin) {
```
Comment (lattmann) [inline src/server/api/index.js]: I know that the old implementation had `!userData.siteAdmin` in the future we may need to use `userData.siteAdmin !== true` to be on the safe side.

**Comment by lattmann** [comment]
Implementation looks ok. See my comments.

File: src/server/api/index.js (hunk 6 above)
Comment (pmeijer) [inline src/server/api/index.js]: Fixed this and fixed the other too.

**Pull request merged**